From 5a8c2283d09f8e628100127f0288a3c022e71ade Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:51:03 -0400
Subject: [tools] perf/top: Use hist_entry__inc_addr_sample

Message-id: <1408953210-26343-176-git-send-email-jolsa@redhat.com>
Patchwork-id: 88957
O-Subject: [PATCH RHEL7.1 BZ1133083 175/322] perf top: Use hist_entry__inc_addr_sample
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 618afb294e288db086fce59de4419644215265e5
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Wed Dec 18 17:08:07 2013 -0300

description
===========
Since it has a hist_entry, no need to skip the hist layer and use the
underlying symbol one.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-top.c b/tools/perf/builtin-top.c
index 73df9c0..2c6cb66 100644
--- a/tools/perf/builtin-top.c
+++ b/tools/perf/builtin-top.c
@@ -190,7 +190,7 @@ static void perf_top__record_precise_ip(struct perf_top *top,
   return;
 
  ip = he->ms.map->map_ip(he->ms.map, ip);
- err = symbol__inc_addr_samples(sym, he->ms.map, counter, ip);
+ err = hist_entry__inc_addr_samples(he, counter, ip);
 
  pthread_mutex_unlock(&notes->lock);
 
-- 
1.7.1